src/languages/swift.js

```diff
@@ -1,96 +1,127 @@
-        'indices insertionSort isBridgedToObjectiveC isBridgedVerbatimToObjectiveC ' +
-        'isUniquelyReferenced isUniquelyReferencedNonObjC join lazy lexicographicalCompare ' +
-        'map max maxElement min minElement numericCast overlaps partition posix ' +
-        'precondition preconditionFailure print println quickSort readLine reduce reflect ' +
-        'reinterpretCast reverse roundUpToAlignment sizeof sizeofValue sort split ' +
-        'startsWith stride strideof strideofValue swap toString transcode ' +
-        'underestimateCount unsafeAddressOf unsafeBitCast unsafeDowncast unsafeUnwrap ' +
-        'unsafeReflect withExtendedLifetime withObjectAtPlusZero withUnsafePointer ' +
-        'withUnsafePointerToObject withUnsafeMutablePointer withUnsafeMutablePointers ' +
-        'withUnsafePointer withUnsafePointers withVaList zip'
-    };
+  // https://docs.swift.org/swift-book/ReferenceManual/LexicalStructure.html#ID413
+  // https://docs.swift.org/swift-book/ReferenceManual/zzSummaryOfTheGrammar.html
+  const DOT_KEYWORD = {
+    className: 'keyword',
+    begin: concat(/\./, lookahead(either(...Swift.dotKeywords, ...Swift.optionalDotKeywords))),
+    end: either(...Swift.dotKeywords, ...Swift.optionalDotKeywords),
+    excludeBegin: true
+  };
+  const KEYWORD_GUARD = {
+    // Consume .keyword to prevent highlighting properties and methods as keywords.
+    begin: concat(/\./, either(...Swift.keywords)),
+    relevance: 0
+  };
+  const KEYWORD = {
+    variants: [
+      {
+        className: 'keyword',
+        begin: either(...Swift.keywords, ...Swift.optionalDotKeywords)
+      },
+      {
+        className: 'literal',
+        begin: either(...Swift.literals)
+      },
+      {
+        className: 'keyword',
+        begin: /\b_\b/,
+        relevance: 0
+      },
+      {
+        className: 'keyword',
+        begin: concat(/#/, either(...Swift.numberSignKeywords))
+      }
+    ]
+  };
+  const KEYWORDS = [
+    DOT_KEYWORD,
+    KEYWORD_GUARD,
+    KEYWORD
+  ];
 
-  const TYPE = {
-    className: 'type',
-    begin: '\\b[A-Z][\\w\u00C0-\u02B8\']*',
+  // https://github.com/apple/swift/tree/main/stdlib/public/core
+  const BUILT_IN_GUARD = {
+    // Consume .built_in to prevent highlighting properties and methods.
+    begin: concat(/\./, either(...Swift.builtIns)),
     relevance: 0
   };
-  // slightly more special to swift
-  const OPTIONAL_USING_TYPE = {
-    className: 'type',
-    begin: '\\b[A-Z][\\w\u00C0-\u02B8\']*[!?]'
+  const BUILT_IN = {
+    className: 'built_in',
+    begin: concat(/\b/, either(...Swift.builtIns), /(?=\()/)
+  };
+  const BUILT_INS = [
+    BUILT_IN_GUARD,
+    BUILT_IN
+  ];
+
+  // https://docs.swift.org/swift-book/ReferenceManual/LexicalStructure.html#ID418
+  const OPERATOR = {
+    className: 'operator',
+    relevance: 0,
+    variants: [
+      {
+        begin: Swift.operator
+      },
+      {
+        begin: `\\.(\\.|${Swift.operatorCharacter})+`
```

Comment:
What are both these cases accomplishing? I'm not sure I understand the second.

Reply:
The first case is a regular operator: a `head` followed by zero or more valid characters.

The second is a dot-operator: only operators that start with a dot are allowed to use dots as characters (`...`, `...<`, `.*`, etc). So there rule here is: a dot followed by one or more characters that may also include dots.